Source/WebKit/Shared/mac/ObjCObjectGraph.mm

```diff
@@ -148,11 +148,7 @@ static bool shouldTransformGraph(id object, const ObjCObjectGraph::Transformer&
         return;
 
     case ObjCType::NSArray: {
-        NSArray *array = object;
-
-        encoder << static_cast<uint64_t>(array.count);
-        for (id element in array)
-            encode(encoder, element);
+        IPC::encode(encoder, static_cast<NSArray *>(object));
```

Comment:
Should be `checked_objc_cast` rather than `static_cast`.

Reply:
Right. The cases handling other types in the same method are all using `static_cast<>()` though, which is why I did the same. Should we convert them too to use a checked ObjectC cast?

Reply:
On second thought, I think that that the logic here is that there is already certainty that the type being casted to is correct, because `typeFromObject()`, used to find out the type of the object, is already using `dynamic_objc_cast<>()`. The checked cast would be redundant, since we are certain that the type is correct, so I will leave the static cast.

Reply:
I am going to change these _all_ to `checked_objc_cast` in a patch later. Your logic about why we don’t need a check applies to _all_ places we use `checked_objc_cast` already. The point is to add an extra check in case our thinking is wrong, to harden for security purposes.

Reply:
Sure, thank you!

Reply:
Actually I took the freedom to submit a PR already, feel free to review. Thanks again! https://github.com/WebKit/WebKit/pull/27219